cylc/flow/cfgspec/workflow.py

```diff
@@ -657,28 +704,29 @@
             other namespaces inherit from it, or a task if no others inherit
             from it.
 
-            Names may not contain colons (which would preclude use of
-            directory paths involving the registration name in ``$PATH``
-            variables). They may not contain the ``.`` character (it will be
-            interpreted as the namespace hierarchy delimiter, separating
-            groups and names -huh?).
+            Names may not contain ``:`` or ``.``.
```

Comment:
```suggestion
            **Names may not contain ``:`` or ``.``.**

Reply:
The task name rules are defined somewhere, should link to that.

Reply:
I've moved the reasoning below this into Cylc-doc